cluster_management/src/main/java/com/pinterest/rocksplicator/Utils.java

```diff
@@ -60,6 +63,40 @@
   private static final String RESOURCE_INFO_META_SUFFIX = "resource_meta";
   private static final String RESOURCE_INFO_CONFIGS_SUFFIX = "resource_configs";
 
+  /**
+   * Obtain the Helix External View of a resource in a cluster.
+   * It performs a fixed number of retries to mitigate potential transient failure.
+   * @param admin Helix Admin client
+   * @param clusterName
+   * @param resourceName
+   * @return ExternalView, which will be null if failed
+   */
+  public static ExternalView getHelixExternalViewWithFixedRetry(HelixAdmin admin, String clusterName, String resourceName) {
+    ExternalView view = admin.getResourceExternalView(clusterName, resourceName);
+    if (view != null) {
+      return view;
+    }
+
+    // retry for maxRetries times with sleep in between
+    final int maxRetries = 3;
+    final int retryWaitSec = 1;
+    try {
+      for (int retryCount = 1; retryCount <= maxRetries; retryCount++) {
+        TimeUnit.SECONDS.sleep(retryWaitSec * retryCount);
+        view = admin.getResourceExternalView(clusterName, resourceName);
+        if (view != null) {
+          return view;
+        }
```

Comment:
Do you want to log that there was an attempt made here to get the external view?

Reply:
good idea, added